### PR TITLE
ref(ts): Convert linkWithConfirmation

### PR DESCRIPTION
**Pull request by leedongwei**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/links/linkWithConfirmation.tsx [hunk 1]
```diff
@@ -4,32 +4,52 @@ import classNames from 'classnames';
 
 import Confirm from 'app/components/confirm';
 
+type Props = {
+  disabled?: boolean;
+  message: React.ReactNode;
+  title: string | undefined;
+  onConfirm: () => void;
+} & React.HTMLAttributes<HTMLAnchorElement>;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/links/linkWithConfirmation.tsx]: Should this take the props of `Confirm`? We don't pass any other props to the anchor

Comment (leedongwei) [reply src/sentry/static/sentry/app/components/links/linkWithConfirmation.tsx]: Fixed to take in props of `Confirm`.

File: src/sentry/static/sentry/app/components/links/linkWithConfirmation.tsx [hunk 2]
```diff
@@ -4,32 +4,48 @@ import classNames from 'classnames';
 
 import Confirm from 'app/components/confirm';
 
+type Props = React.ComponentProps<typeof Confirm> &
+  React.HTMLAttributes<HTMLAnchorElement>;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/links/linkWithConfirmation.tsx]: Why also anchor?

Comment (leedongwei) [reply src/sentry/static/sentry/app/components/links/linkWithConfirmation.tsx]: We have an `<a>` element inside `Confirm` and it requires some props specific to anchors.

Comment (leedongwei) [reply src/sentry/static/sentry/app/components/links/linkWithConfirmation.tsx]: Good suggestions Mr Evan.

**Review by evanpurkhiser — APPROVED** [review]

**Comment by leedongwei** [comment]
Rebased against `master` and force-pushed because Travis was a bit too flakey.

**Review by markstory — APPROVED** [review]

**Comment by billyvg** [comment]
@leedongwei We converted `<Link>` alongside `<LinkWithConfirmation>`, are there any changes you want to pull in from this PR? If not can we close?

**Pull request closed without merging**